File: Algorithms/Sorting Algorithm/1 Selection sort/SelectionSort.py
def selectionSort(arr,size):

    for i in range(size-1):
        
        min = i

        for j in range(i+1,size):
            if arr[j] < arr[min]:
                min = j 
        
        arr[i],arr[min] = arr[min],arr[i]

x = [1,6,3,5,-84,-6,-7,-3,-2,0]

selectionSort(x,len(x))

print(x)

File: Algorithms/Sorting Algorithm/1 Selection sort/SelectionSort.cpp

#include <iostream>
using namespace std;

int SelectionSort(int arr[],int n){
    
    for(int i = 0 ; i < n-1; i++){
        int min = i;
        
        for(int j = i+1; j < n ; j++){
            if (arr[j] < arr[min]){
                
                min = j;
            }
        }
        
        swap(arr[i],arr[min]);
    }
    return 0;
}

int main()
{
    int n;
    cout<<"Enter the number of Element: ";
    cin>> n;
    int arr[n];
    
    for(int i = 0;i<n;i++){
        
        int x;
        cout<<"Enter the element: ";
        cin >> x;
        arr[i] = x;
    }
    
    SelectionSort(arr,n);
    
    for(int i = 0;i<n;i++){
        cout<<arr[i];
        cout<<" ";
    }

    return 0;
}

